### PR TITLE
[17.0][MIG] sale_start_end_dates

### PR DESCRIPTION
**Pull request by alexis-via**
Migration from v16 to v17. Code cleanup in tests: stop using demo data, use setupClass(cls)

### CODE REVIEW

File: sale_start_end_dates/tests/test_sale_start_end_dates.py [hunk 1]
```diff
@@ -8,40 +8,47 @@
...
 class TestSaleStartEndDates(TransactionCase):
-    def setUp(self):
-        super().setUp()
-        self.partner = self.env.ref("base.res_partner_3")
-        self.product_id = self.env.ref(
-            "account_invoice_start_end_dates.product_insurance_contract_demo"
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```
Comment (rousseldenis) [inline sale_start_end_dates/tests/test_sale_start_end_dates.py:14]: @alexis-via FYI, best is to use BaseCommon class instead : https://github.com/odoo/odoo/blob/c11e5fea5dd5a357c17b12c0c5f383909fa44bb8/odoo/addons/base/tests/common.py#L21

**Review by rousseldenis — APPROVED** [review]
Code review

File: sale_start_end_dates/tests/test_sale_start_end_dates.py (hunk 1 above)
Comment (alexis-via) [reply sale_start_end_dates/tests/test_sale_start_end_dates.py:14]: I don't think so. BaseCommon adds stuff for the tests of the base module. Nobody uses this class for tests except the product module. I didn't find any OCA module using it... do you know one ?

**Review by StefanRijnhart — APPROVED** [review]

**Pull request merged**